File: (P) Practice/CP8-111502036/P8-111502036/P8-111502036.cpp
/*
Practice 8 C++
Name: 沈哲寬
Student Number: 111502036
Course 2022-CE1003-B
*/
#include<iostream>
using namespace std;
int gcd(int a, int b){
    if(b==0){
        return a;
    }
    return gcd(b,a%b);
}
int main(){
    int number1,number2;
    cin >> number1 >> number2;
    cout << gcd(number1,number2)<<endl;
}

File: (P) Practice/P8-111502036/P8-111502036/P8-111502036.py
"""
Practice 8 python
Name: 沈哲寬
Student Number: 111502036
Course 2022-CE1003-B
"""
def preprocess():
    with open('input.txt', 'r') as file:
        data = list(map(int, file.read().split()))
    return data
'''
def compare(a,b):
    if(a>b):
        temp = a
        a = b
        b = temp
    return
'''
def bubble_sort(List):
    n = len(List)
    for j in range(n-1):
        for i in range(n-1):
            if(List[i]>List[i+1]):
                temp = List[i]
                List[i] = List[i+1]
                List[i+1] = temp
    return List
List = preprocess()
print(bubble_sort(List))